applications/CoSimulationApplication/python_scripts/base_classes/co_simulation_io.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import print_function, absolute_import, division  # makes these scripts backward compatible with python 2.6 and 2.7
+
+##
+#  IMPORTANT : This is a BASE CLASS
+#               Please do not change any thing in this class.
+#
+# This Class servers as a base class for all the Input-output methods to be implemented
+class CoSimulationIO(object):
+    ## The constructor
+    #
+    def __init__(self, model, settings):
+        self.echo_level = 0
+        self.settings = settings
+        self.model = model
+
+    ## ImportCouplingInterfaceData :  used to import data from other solvers
+    #                Follow EXAMPLE implementation below.
+    #
+    #  @param data_object     python dictionary : object of the data to be imported.
+    #                                             data will be imported into this dictionary
+    #  @param from_solver     python object : The solver from which data is to be imported.
+    def ImportCouplingInterfaceData(self, data_object, from_solver=None):
+        """
+        if(from_client not None):
+            # exchange data from python cosim solver
+        else:
+            # import data from remote solver
+        """
+        raise NotImplementedError(cs_tools.darkred("From BaseIO : The method ImportCouplingInterfaceData is not implemented in the IO class!"))
```

Comment:
Where is this `cs_tools` module imported?

Reply:
will be solved in the future, we are currently not sure how this class will be used